service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RespondentController.java

```diff
@@ -117,4 +149,162 @@ public void handleSubmitted(@RequestBody CallbackRequest callbackRequest) {
+
+        OtherToRespondentEventData eventData = caseData.getOtherToRespondentEventData();
+        Other selectedPreparedOther = othersService.getSelectedPreparedOther(caseData, eventData.getOthersList())
+            .getValue();
+        caseDetails.getData().put(TRANSFORMED_RESPONDENT,
+            respondentService.transformOtherToRespondent(selectedPreparedOther));
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/change-from-other/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleChangeFromOtherMidEvent(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+        OtherToRespondentEventData eventData = caseData.getOtherToRespondentEventData();
+
+        List<Element<Respondent>> newRespondents = confidentialDetailsService.prepareCollection(
+            caseData.getAllRespondents(), caseData.getConfidentialRespondents(), expandCollection());
+        newRespondents.add(newElement(eventData.getTransformedRespondent()));
+
+        CaseData dummyCaseData = caseData.builder()
+            .respondents1(newRespondents)
+            .build();
+
+        List<String> errors = respondentValidator.validate(dummyCaseData, dummyCaseData, true);
+        return respond(caseDetails, errors);
+    }
+
+    private UUID getFirstOtherId(CaseData caseData) {
+        Set<UUID> additionalOtherIds = nullSafeList(caseData.getOthers().getAdditionalOthers())
+            .stream().map(Element::getId).collect(Collectors.toSet());
+        UUID firstOtherUUID = caseData.getConfidentialOthers().stream().map(Element::getId)
+            .filter(co -> !additionalOtherIds.contains(co)).findFirst()
+            .orElse(UUID.fromString("00000000-0000-0000-0000-000000000000"));
+        return firstOtherUUID;
+    }
+
+    private Element<Other> getSelectedOther(CaseData caseData) {
+        UUID firstOtherUUID = getFirstOtherId(caseData);
+        return othersService.getSelectedOther(caseData,
+            caseData.getOtherToRespondentEventData().getOthersList(), firstOtherUUID);
+    }
+
+    private List<Element<Other>> buildNewAllOthers(CaseData caseData, Element<Other> selectedOther) {
+        List<Element<Other>> listOfNewOtherElement = new ArrayList<>(caseData.getAllOthers());
+        listOfNewOtherElement.removeIf(ele -> Objects.equals(ele.getValue(), selectedOther.getValue()));
+        List<Element<Other>> ret = new ArrayList<>();
+        final List<Element<Other>> confidentialOthers = caseData.getConfidentialOthers();
+        listOfNewOtherElement.forEach(element -> {
+            if (element.getValue().containsConfidentialDetails()) {
+                Element<Other> otherElementWithConfidentialDetails = element(element.getId(), othersService
+                    .addConfidentialDetails(getConfidentialItemToAdd(confidentialOthers, element), element));
```

Comment:
check why we need to addConfidentailDetails here. do we perhaps just need to remove selected other from confidentialOthers?